src/controller/AutoCommissioner.cpp

```diff
@@ -139,24 +141,47 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStage(CommissioningStag
         // operational network because the provisioning of certificates will trigger the device to start operational advertising.
         if (mNeedsNetworkSetup)
         {
-            if (mParams.GetWiFiCredentials().HasValue() &&
-                mNetworkTechnology.Has(
-                    chip::app::Clusters::NetworkCommissioning::NetworkCommissioningFeature::kWiFiNetworkInterface))
+            if (!mNetworkTechnology.HasAny(NetworkCommissioningFeature::kWiFiNetworkInterface,
+                                           NetworkCommissioningFeature::kThreadNetworkInterface))
             {
-                return CommissioningStage::kWiFiNetworkSetup;
+                ChipLogError(Controller, "Device does NOT support ANY network setup (thread or wifi)");
             }
-            else if (mParams.GetThreadOperationalDataset().HasValue() &&
-                     mNetworkTechnology.Has(
-                         chip::app::Clusters::NetworkCommissioning::NetworkCommissioningFeature::kThreadNetworkInterface))
+
+            if (mNetworkTechnology.Has(NetworkCommissioningFeature::kWiFiNetworkInterface))
             {
-                return CommissioningStage::kThreadNetworkSetup;
+                ChipLogProgress(Controller, "Device supports WiFi");
+                if (mParams.GetWiFiCredentials().HasValue())
+                {
+                    ChipLogProgress(Controller, "Setting up WiFi configuration");
+                    return CommissioningStage::kWiFiNetworkSetup;
+                }
+                else
+                {
+                    ChipLogProgress(Controller, "WiFi parameters not available");
+                }
             }
-            else
+            if (mNetworkTechnology.Has(NetworkCommissioningFeature::kWiFiNetworkInterface))
             {
-                ChipLogError(Controller, "Required network information not provided in commissioning parameters");
```

Comment:
This code section is being revised in https://github.com/project-chip/connectedhomeip/pull/14567, which obsolete this change fairly quickly. Instead, could this error message just be changed to report what commissioning parameters were supplied vs. what is supported in the cluster? 
Ex - Required network information not provided - %s parameters were supplied, device network commissioning cluster supports %s"

Reply:
I re-updated based on master.